src/ResourceManager/OperationalInsights/Commands.OperationalInsights/Client/OperationalInsightsClient.DataSources.cs

```diff
@@ -0,0 +1,209 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Hyak.Common;
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.Commands.OperationalInsights.Properties;
+using Microsoft.Azure.Management.OperationalInsights;
+using Microsoft.Azure.Management.OperationalInsights.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Client
+{
+    public partial class OperationalInsightsClient
+    {
+        public virtual PSDataSource GetDataSource(string resourceGroupName, string workspaceName, string dataSourceName)
+        {
+            var response = OperationalInsightsManagementClient.DataSources.Get(resourceGroupName, workspaceName, dataSourceName);
+
+            return new PSDataSource(response.DataSource, resourceGroupName, workspaceName);
+        }
+
+        public virtual List<PSDataSource> ListDataSources(string resourceGroupName, string workspaceName, string kind)
+        {
+            List<PSDataSource> dataSources = new List<PSDataSource>();
+            
+            // List data sources by kind
+            var response = OperationalInsightsManagementClient.DataSources.ListInWorkspace(resourceGroupName, workspaceName, kind, string.Empty);
+            if (response != null && response.DataSources != null)
```

Comment:
These entire code can be simplified a lot. You could remove lines 46 through 58 and have following code there
```
while (null!=response) 
{
                if (response.DataSources != null)
                {
                    response.DataSources.ForEach(si => dataSources.Add(new PSDataSource(si, resourceGroupName, workspaceName)));
                }
                if(!string.IsNullOrEmpty(response.NextLink))
                {
                    response = OperationalInsightsManagementClient.DataSources.ListNext(response.NextLink);
                 }
                else
                {
                           break;
                }
            }   
```